app/one_click_unsubscribe/rest.py

```diff
@@ -52,3 +60,62 @@ def get_unsubscribe_request_data(notification, email_address):
         "service_id": notification.service_id,
         "email_address": email_address,
     }
+
+
+def create_unsubscribe_request_reports_summary(service_id):
+    reports_summary = []
+    batched_unsubscribe_reports_summary = []
+    unbatched_unsubscribe_report_summary = []
+    # Check for existing unsubscribe reports and create their summaries
+    if unsubscribe_request_reports := get_unsubscribe_request_reports_dao(service_id):
+        batched_unsubscribe_reports_summary = _create_batched_unsubscribe_request_reports_summary(
+            unsubscribe_request_reports,
+        )
+        batched_unsubscribe_reports_summary = batched_unsubscribe_reports_summary
+    # Check for unsubscribe requests and create a summary for them
+    if unbatched_unsubscribe_requests := get_unbatched_unsubscribe_requests_dao(service_id):
+        if batched_unsubscribe_reports_summary:
+            earliest_timestamp = batched_unsubscribe_reports_summary[0]["latest_timestamp"]
+        else:
+            service = dao_fetch_service_by_id(service_id)
+            earliest_timestamp = service.created_at
+
+        unbatched_unsubscribe_report_summary.append(
+            _create_unbatched_unsubscribe_request_report_summary(unbatched_unsubscribe_requests, earliest_timestamp)
+        )
+
+    reports_summary = unbatched_unsubscribe_report_summary + batched_unsubscribe_reports_summary
+
+    return reports_summary
+
+
+def _create_batched_unsubscribe_request_reports_summary(unsubscribe_request_reports: list) -> list:
+    report_summaries = []
+    for report in unsubscribe_request_reports:
+        report_summary = {
+            "batch_id": str(report.id),
+            "count": report.count,
+            "earliest_timestamp": report.earliest_timestamp,
+            "latest_timestamp": report.latest_timestamp,
+            "processed_by_service_at": report.processed_by_service_at,
+            "is_a_batched_report": True,
+            "status": "Completed" if report.processed_by_service_at else "Downloaded",
+        }
+        report_summaries.append(report_summary)
+    return report_summaries
+
+
+def _create_unbatched_unsubscribe_request_report_summary(
+    unbatched_unsubscribe_requests: list, earliest_timestamp: str = None
+) -> dict:
+    count = len(unbatched_unsubscribe_requests)
+    report_summary = {
+        "batch_id": str(uuid.uuid4()),
```

Comment:
I’m a bit unclear how this will work, because the API will return a different ID on each request. That means the admin app won’t have a fixed way to ask for the unbatched requests. I think the admin app can work that out by looking at `is_a_batched_report` instead, but it’s confusing to return a UUID here instead of `None`:
```suggestion
        "batch_id": None,
```

Reply:
Good catch. You are right, it should be None.